.github/workflows/build_dev_docs.yml

```diff
@@ -0,0 +1,57 @@
+name: Build Dev Docs
+
+on:
+  push:
+    branches:
+      - dev
+
+jobs:
+  build-docs:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2 
+      with:
+          submodules: 'recursive'
+    - name: Set up Python 
+      uses: actions/setup-python@v2
+      with:
+          python-version: 3.8 
+    - name: Build python-dp
+      run: |
+        wget https://github.com/bazelbuild/bazel/releases/download/2.1.0/bazel-2.1.0-installer-linux-x86_64.sh
+        chmod +x bazel-2.1.0-installer-linux-x86_64.sh
+        ./bazel-2.1.0-installer-linux-x86_64.sh --user
+        export PATH=/github/home/bin:$PATH
+        rm bazel-2.1.0-installer-linux-x86_64.sh
+        pip install pipenv
+        ./build_PyDP.sh
+         
+   
+    - uses: ammaraskar/sphinx-action@master
+      with:
+        pre-build-command: "python -m pip install -U pip setuptools wheel && python setup.py bdist_wheel && python -m pip install --ignore-installed dist/*.whl"
+        docs-folder: "docs/"
+   
+    # Publish built docs to gh-pages branch.
+    # ===============================
+    - name: Commit documentation changes
+      run: |
+        git clone https://github.com/ammaraskar/sphinx-action-test.git --branch gh-pages --single-branch gh-pages
+        cp -r docs/_build/html/* gh-pages/
+        cd gh-pages
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+        git add .
+        git commit -m "Update documentation" -a || true
+        # The above command will fail if no changes were present, so we ignore
+        # that.
+    - name: Push changes
+      uses: ad-m/github-push-action@master
+      with:
+        branch: gh-pages
+        directory: gh-pages
+        force: true
+        github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Which token needs to be added here?

Reply:
That gets picked up by default. No need to configure it.